.gitlab-ci.yml

```diff
@@ -72,24 +72,36 @@ check:nix-dry:
     - if: $CI_COMMIT_BRANCH && $CI_COMMIT_BRANCH != 'master' && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
       when: manual
 
-check:test:
+check:test-generate:
   stage: check
-  needs: []
   script:
+    - mkdir -p ./tmp
     - >
         nix-shell --run '
-        npm run build --verbose;
-        npm test -- --ci --coverage;
+        ./scripts/test-pipelines.sh > ./tmp/test-pipelines.yml
         '
   artifacts:
     when: always
-    reports:
-      junit:
-        - ./tmp/junit/junit.xml
-      coverage_report:
-        coverage_format: cobertura
-        path: ./tmp/coverage/cobertura-coverage.xml
-  coverage: '/All files[^|]*\|[^|]*\s+([\d\.]+)/'
+    paths:
+      - ./tmp/test-pipelines.yml
+  rules:
+    # Runs on staging commits and ignores version commits
+    - if: $CI_COMMIT_BRANCH =~ /^feature.*$/ && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
+    # Manually run on commits other than master and staging and ignore version commits
+    - if: $CI_COMMIT_BRANCH && $CI_COMMIT_BRANCH !~ /^(?:master|staging)$/ && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
+      when: manual
+
+check:test:
+  stage: check
+  needs:
+    - check:test-generate
+  inherit:
+    variables: false
+  trigger:
+    include:
+      - artifact: tmp/test-pipelines.yml
+        job: check:test-generate
+    strategy: depend
   rules:
     # Runs on staging commits and ignores version commits
     - if: $CI_COMMIT_BRANCH =~ /^feature.*$/ && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
```

Comment:
I'm not sure if `rules` here are necessary because it is a generated pipeline.

That is, it only triggers if the parent job which is `check:test-generate` runs, if that doesn't run, there's no `tmp/test-pipelines.yml`, and therefore, it shouldn't run anyway.

You should try if it works by removing the `rules` section under `check:test`.

Reply:
Without the rules there gitlab says the yaml is incorrect with the error `'check:test' job needs 'check:test-generate' job, but 'check:test-generate' is not in any previous stage`

Reply:
Did you remove the rules from `check:test-generate` or `check:test`? Only remove rules from `check:test`.

If that doesn't work, then I guess rules must exist for these 2 jobs, but child pipeline jobs shouldn't require rules or workflow.

Reply:
I only removed the rues from `check:test`, so I think the rules are required there.